src/main/java/jssc/SerialNativeInterface.java

```diff
@@ -98,6 +98,12 @@ else if(osName.equals("Mac OS X") || osName.equals("Darwin"))
         }
     }
 
+    /**
+     * Default constructor
+     * TODO: This class is effectively static, why instantiate it?
```

Comment:
Make the contructor private and comply with https://rules.sonarsource.com/java/tag/design/RSPEC-1118

Reply:
It can't without refactoring and the refactoring breaks the JNI calls.  This is because some JNI functions are not `static`.  Search for `new SerialNativeInterface()`.